src/Microsoft.VisualStudio.Services.Agent/AgentServer.cs

```diff
@@ -86,10 +86,10 @@ private void CheckConnection()
         // Configuration
         //-----------------------------------------------------------------
 
-        public Task<List<TaskAgentPool>> GetAgentPoolsAsync(string agentPoolName)
+        public Task<List<TaskAgentPool>> GetAgentPoolsAsync(string agentPoolName, TaskAgentPoolType poolType = TaskAgentPoolType.Automation)
         {
             CheckConnection();
-            return _taskAgentClient.GetAgentPoolsAsync(agentPoolName);
+            return _taskAgentClient.GetAgentPoolsAsync(agentPoolName, poolType: poolType);
```

Comment:
why don't we just change the server to return both pool type?

Reply:
currently we dont have any consumer which requires all type of pool. I would like to keep it tightly bound with type.